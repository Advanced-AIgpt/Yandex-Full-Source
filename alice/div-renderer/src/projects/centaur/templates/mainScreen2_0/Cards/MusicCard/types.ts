import { IAbstractCardProps } from '../types';

export interface IMusicCardProps extends IAbstractCardProps {
    type: 'music';
    name?: string | null;
    description?: string | null;
    color?: string | null;
    cover?: string | null;
    modified?: string | null;
}
