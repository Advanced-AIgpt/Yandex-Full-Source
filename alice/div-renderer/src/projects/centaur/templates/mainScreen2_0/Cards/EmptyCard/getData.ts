import { IEmptyCardProps } from './types';
import { ICardDataAdapter } from '../types';

export const getEmptyCardData: ICardDataAdapter<IEmptyCardProps> = function getEmptyCardData(card, requestState) {
    if (card && card.VacantCardData) {
        return {
            type: 'empty',
            requestState,
        };
    }
    return null;
};
