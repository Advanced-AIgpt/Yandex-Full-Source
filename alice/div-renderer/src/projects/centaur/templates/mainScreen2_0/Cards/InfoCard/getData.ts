import { sample } from 'lodash';
import { IInfoCardProps } from './types';
import { DISCOVERY_CARDS } from '../../../mainScreen/discovery';
import { ICardDataAdapter } from '../types';
import { textAction } from '../../../../../../common/actions';

export const getInfoCardData: ICardDataAdapter<IInfoCardProps> = function getInfoCardData(card, requestState) {
    if (typeof card.InfoCardData !== 'undefined' && card.InfoCardData !== null) {
        const data = card.InfoCardData.Title === 'test' ? DISCOVERY_CARDS[0] : sample(DISCOVERY_CARDS) || DISCOVERY_CARDS[0];

        return {
            type: 'info',
            title: data.title,
            image_background: data.bg,
            description: data.description,
            requestState,
            actions: [
                {
                    log_id: 'discovery.mainscreen.execute',
                    url: textAction(data.description),
                },
            ],
        };
    }
    return null;
};
