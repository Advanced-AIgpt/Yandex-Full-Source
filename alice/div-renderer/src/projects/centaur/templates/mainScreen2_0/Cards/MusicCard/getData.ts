import { IMusicCardProps } from './types';
import { colorDarkGrey } from '../../../../style/constants';
import { ICardDataAdapter } from '../types';

const playlists: readonly [string, string][] = [
    ['Плейлист дня', '#31AD54'],
    ['Мне нравится', '#E53866'],
    ['Тайник', '#186FD3'],
    ['Подкасты', '#036292'],
    ['Дежавю', '#A019D1'],
    ['Премьера', '#D1673E'],
    ['Плейлист с Алисой', '#2D0B87'],
    ['Мой 2021', '#C20146'],
    ['Плейлист семьи', '#FE8853'],
];

export const getMusicCardData: ICardDataAdapter<IMusicCardProps> = function getMusicCardData(card, requestState) {
    if (typeof card.MusicCardData !== 'undefined' && card.MusicCardData !== null) {
        const data = card.MusicCardData;

        const knownPlaylist = playlists.find(el => el[0] === data.Name);

        const color = knownPlaylist ? knownPlaylist[1] : colorDarkGrey;

        return {
            type: 'music',
            name: data.Name,
            modified: data.Modified?.value,
            color,
            cover: data.Cover,
            actions: (card.Action && [
                {
                    log_id: 'music_card_action',
                    url: card.Action,
                },
            ]) || undefined,
            preventDefaultLoader: true,
            requestState,
        };
    }
    return null;
};
