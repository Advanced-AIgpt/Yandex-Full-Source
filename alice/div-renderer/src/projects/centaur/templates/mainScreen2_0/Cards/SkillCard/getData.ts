import { ISkillCardProps } from './types';
import { ICardDataAdapter } from '../types';
import { NAlice } from '../../../../../../protos';
import { EnumLayer } from '../../../../actions/client';
type ITButton = NAlice.NData.TCentaurWidgetCardData.TExternalSkillCardData.TMainScreenData.ITButton;

interface INormalisedButton {
    Text: string;
    Payload?: string;
}

function isNormalButton(el: ITButton): el is INormalisedButton {
    return Boolean(el.Text);
}

export const getSkillCardData: ICardDataAdapter<ISkillCardProps> = function getSkillCardData(card, requestState) {
    if (typeof card.ExternalSkillCardData !== 'undefined' && card.ExternalSkillCardData !== null) {
        const data = card.ExternalSkillCardData;

        if (data.widgetGalleryData) {
            return {
                type: 'skill',
                requestState,
                skillName: data.skillInfo?.Name || ' ',
                skillImage: data.skillInfo?.Logo || ' ',
                skillId: data.skillInfo?.SkillId || ' ',
                title: ' ',
            };
        } else {
            return {
                type: 'skill',
                requestState,
                title: data.mainScreenData?.Title,
                comment: data.mainScreenData?.Text,
                image: data.mainScreenData?.ImageUrl,
                skillName: data.skillInfo?.Name || ' ',
                skillImage: data.skillInfo?.Logo || ' ',
                skillId: data.skillInfo?.SkillId || ' ',
                layer: EnumLayer.dialog,
                buttons: data.mainScreenData?.buttons
                    ?.filter(isNormalButton)
                    .map(el => ({
                        text: el.Text,
                        payload: el.Payload || undefined,
                    })),
            };
        }
    }
    return null;
};
