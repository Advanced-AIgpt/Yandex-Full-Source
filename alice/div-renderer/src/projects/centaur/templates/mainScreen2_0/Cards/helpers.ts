export function getCardMainScreenId({ rowIndex, colIndex }: { rowIndex?: number; colIndex?: number }) {
    return `main_screen_card_${rowIndex || 0}_${colIndex || 0}`;
}
