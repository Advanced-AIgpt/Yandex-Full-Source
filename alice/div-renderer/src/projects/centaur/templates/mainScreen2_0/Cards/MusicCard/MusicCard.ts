import {
    ContainerBlock,
    Div,
    FixedSize,
    ImageBlock,
    MatchParentSize,
    SolidBackground,
    TextBlock,
} from 'divcard2';
import { compact } from 'lodash';
import { BasicCard } from '../BasicCard';
import { text28m, title32m, title48m } from '../../../../style/Text/Text';
import { colorWhiteOpacity10, colorWhiteOpacity50 } from '../../../../style/constants';
import EmptyDiv from '../../../../components/EmptyDiv';
import { IMusicCardProps } from './types';
import { logger } from '../../../../../../common/logger';
import { updatedAt } from '../../../../helpers/helpers';
import { getS3Asset } from '../../../../helpers/assets';
import { musicLoaderElement } from '../../../Music/MusicLoader/MusicLoader';
import { getCardMainScreenId } from '../helpers';

function MusicErrorCard({ rowIndex, colIndex }: {rowIndex?: number; colIndex?: number}): Div {
    return BasicCard({
        id: getCardMainScreenId({ rowIndex, colIndex }),
        items: [
            new TextBlock({
                ...title32m,
                text_color: colorWhiteOpacity50,
                text: 'Музыка',
            }),
            new EmptyDiv({
                height: new MatchParentSize({ weight: 2 }),
            }),
            new ContainerBlock({
                content_alignment_horizontal: 'center',
                items: [
                    new TextBlock({
                        ...title48m,
                        width: new FixedSize({ value: 135 }),
                        height: new FixedSize({ value: 135 }),
                        text_alignment_horizontal: 'center',
                        text_alignment_vertical: 'center',
                        line_height: 0,
                        text: '✕',
                        background: [
                            new SolidBackground({ color: colorWhiteOpacity10 }),
                        ],
                        border: {
                            corner_radius: 67,
                        },
                    }),
                    new TextBlock({
                        ...title32m,
                        margins: {
                            top: 24,
                        },
                        text_color: colorWhiteOpacity50,
                        text_alignment_horizontal: 'center',
                        text: 'Что-то пошло не так',
                    }),
                    new TextBlock({
                        ...text28m,
                        margins: {
                            top: 8,
                        },
                        text_color: colorWhiteOpacity50,
                        text_alignment_horizontal: 'center',
                        text: 'Проверьте интернет или обновите экран',
                    }),
                ],
            }),
            new EmptyDiv({
                height: new MatchParentSize({ weight: 3 }),
            }),
        ],
    });
}

export default function MusicCard({
    color,
    name,
    description,
    cover,
    modified,
    actions,
    longtap_actions,
    requestState,
    rowIndex,
    colIndex,
}: Partial<IMusicCardProps>): Div {
    if (!name || !color || (!description && !modified) || !cover) {
        logger.error('The music card should have a name, color, description and cover, but some of this was not transmitted');

        return MusicErrorCard({
            rowIndex,
            colIndex,
        });
    }

    const cardData: Parameters<typeof BasicCard>[0] = {
        id: getCardMainScreenId({ rowIndex, colIndex }),
        background: [new SolidBackground({ color })],
        items: [
            new TextBlock({
                ...title32m,
                text_color: colorWhiteOpacity50,
                text: 'Рекомендуем',
            }),
            new EmptyDiv({
                height: new MatchParentSize(),
            }),
            new ImageBlock({
                image_url: cover,
                height: new FixedSize({ value: 220 }),
                width: new FixedSize({ value: 220 }),
                alignment_horizontal: 'center',
                border: {
                    corner_radius: 18,
                },
            }),
            new TextBlock({
                ...title32m,
                text_alignment_horizontal: 'center',
                margins: {
                    top: 24,
                },
                text: name,
            }),
            new TextBlock({
                ...title32m,
                text_color: colorWhiteOpacity50,
                text_alignment_horizontal: 'center',
                margins: {
                    top: 4,
                },
                text: description || updatedAt(new Date(modified ?? 0)),
            }),
            new EmptyDiv({
                height: new MatchParentSize(),
            }),
            new ImageBlock({
                image_url: getS3Asset('player/play.png?v=1'),
                height: new FixedSize({ value: 79 }),
                width: new FixedSize({ value: 79 }),
                margins: {
                    bottom: 8,
                },
                alignment_horizontal: 'center',
            }),
        ],
        actions: actions ? compact([
            requestState && musicLoaderElement({
                title: name,
                subtitle: description,
                ImageUrl: cover,
            }, requestState),
            ...actions,
        ]) : undefined,
        longtap_actions,
    };

    return BasicCard(cardData);
}
