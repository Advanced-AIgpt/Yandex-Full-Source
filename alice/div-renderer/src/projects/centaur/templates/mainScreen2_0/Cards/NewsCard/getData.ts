import { compact } from 'lodash';
import { INewsCardProps } from './types';
import { EnumLayer } from '../../../../actions/client';
import { getImageByTopic } from '../../../../helpers/newsHelper';
import { ICardDataAdapter } from '../types';

export const getNewsCardData: ICardDataAdapter<INewsCardProps> = function getNewsCardData(card, requestState) {
    if (typeof card.NewsCardData !== 'undefined' && card.NewsCardData !== null) {
        const data = card.NewsCardData;

        return {
            type: 'news',
            title: data.Title,
            content: data.Content,
            image: data.ImageUrl || getImageByTopic(data.Topic) || undefined,
            layer: EnumLayer.dialog,
            requestState,
            actions: compact([
                card.Action && {
                    log_id: 'news_card_action',
                    url: card.Action,
                },
            ]),
        };
    }
    return null;
};
