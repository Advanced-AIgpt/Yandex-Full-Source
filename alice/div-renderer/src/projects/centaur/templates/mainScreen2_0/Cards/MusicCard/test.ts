import MusicCard from './MusicCard';
import { AnonymizeDataForSnapshot } from '../../../../../../common/helpers/dev';
import { getMusicCardData } from './getData';
import { NAlice } from '../../../../../../protos';
import { logger } from '../../../../../../common/logger';
import { createRequestState } from '../../../../../../registries/common';
const TCardData = NAlice.NData.TCentaurMainScreenMyScreenData.TColumn.TCardData;

jest.mock('../../../../../../common/logger');

describe('Main screen empty card', () => {
    it('should get actual data', () => {
        expect(getMusicCardData(new TCardData({}), createRequestState())).toBeNull();

        expect(getMusicCardData(new TCardData({
            YouTubeCardData: {},
        }), createRequestState())).toBeNull();

        expect(getMusicCardData(new TCardData({
            YouTubeCardData: {},
            MusicCardData: {
            },
        }), createRequestState())).not.toBeNull();

        expect(getMusicCardData(new TCardData({
            MusicCardData: {
                Name: 'Плейлист дня',
                Modified: {
                    value: 'Обновлен сегодня',
                },
                Cover: '1',
                Color: '#31AD54',
            },
        }), createRequestState())).not.toBeNull();
    });

    it('should match snapshot', () => {
        expect(AnonymizeDataForSnapshot(MusicCard({
            type: 'music',
            name: 'Плейлист дня',
            color: '#31AD54',
            description: 'Обновлен сегодня',
            cover: 'link_to_image',
        }))).toMatchSnapshot();

        expect(logger.error).toHaveBeenCalledTimes(0);
    });

    it('should match snapshot with error', () => {
        expect(AnonymizeDataForSnapshot(MusicCard({
            type: 'music',
            name: '',
            color: '#31AD54',
            description: 'Обновлен сегодня',
            cover: 'link_to_image',
        }))).toMatchSnapshot();

        expect(logger.error).toHaveBeenCalled();
    });
});
