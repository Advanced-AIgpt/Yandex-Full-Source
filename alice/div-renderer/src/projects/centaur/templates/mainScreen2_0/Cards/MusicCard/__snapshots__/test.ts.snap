// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main screen empty card should match snapshot 1`] = `
Object {
  "background": Array [
    Object {
      "color": "#31AD54",
      "type": "solid",
    },
  ],
  "border": Object {
    "corner_radius": 28,
  },
  "height": Object {
    "type": "match_parent",
    "weight": 1,
  },
  "id": "main_screen_card_0_0",
  "items": Array [
    Object {
      "font_size": 32,
      "font_weight": "medium",
      "line_height": 40,
      "text": "Рекомендуем",
      "text_color": "#80ffffff",
      "type": "text",
    },
    Object {
      "delimiter_style": Object {
        "color": "#0fff",
      },
      "height": Object {
        "type": "match_parent",
      },
      "type": "separator",
    },
    Object {
      "alignment_horizontal": "center",
      "border": Object {
        "corner_radius": 18,
      },
      "height": Object {
        "type": "fixed",
        "value": 220,
      },
      "image_url": "link_to_image",
      "type": "image",
      "width": Object {
        "type": "fixed",
        "value": 220,
      },
    },
    Object {
      "font_size": 32,
      "font_weight": "medium",
      "line_height": 40,
      "margins": Object {
        "top": 24,
      },
      "text": "Плейлист дня",
      "text_alignment_horizontal": "center",
      "text_color": "#fff",
      "type": "text",
    },
    Object {
      "font_size": 32,
      "font_weight": "medium",
      "line_height": 40,
      "margins": Object {
        "top": 4,
      },
      "text": "Обновлен сегодня",
      "text_alignment_horizontal": "center",
      "text_color": "#80ffffff",
      "type": "text",
    },
    Object {
      "delimiter_style": Object {
        "color": "#0fff",
      },
      "height": Object {
        "type": "match_parent",
      },
      "type": "separator",
    },
    Object {
      "alignment_horizontal": "center",
      "height": Object {
        "type": "fixed",
        "value": 79,
      },
      "image_url": "https://dialogs.s3.yandex.net/smart_displays/player/play.png?v=1",
      "margins": Object {
        "bottom": 8,
      },
      "type": "image",
      "width": Object {
        "type": "fixed",
        "value": 79,
      },
    },
  ],
  "paddings": Object {
    "bottom": 24,
    "left": 24,
    "right": 24,
    "top": 24,
  },
  "type": "container",
  "width": Object {
    "type": "match_parent",
    "weight": 1,
  },
}
`;

exports[`Main screen empty card should match snapshot with error 1`] = `
Object {
  "background": Array [
    Object {
      "color": "#1aeaf6ec",
      "type": "solid",
    },
  ],
  "border": Object {
    "corner_radius": 28,
  },
  "height": Object {
    "type": "match_parent",
    "weight": 1,
  },
  "id": "main_screen_card_0_0",
  "items": Array [
    Object {
      "font_size": 32,
      "font_weight": "medium",
      "line_height": 40,
      "text": "Музыка",
      "text_color": "#80ffffff",
      "type": "text",
    },
    Object {
      "delimiter_style": Object {
        "color": "#0fff",
      },
      "height": Object {
        "type": "match_parent",
        "weight": 2,
      },
      "type": "separator",
    },
    Object {
      "content_alignment_horizontal": "center",
      "items": Array [
        Object {
          "background": Array [
            Object {
              "color": "#1affffff",
              "type": "solid",
            },
          ],
          "border": Object {
            "corner_radius": 67,
          },
          "font_size": 48,
          "font_weight": "medium",
          "height": Object {
            "type": "fixed",
            "value": 135,
          },
          "line_height": 0,
          "text": "✕",
          "text_alignment_horizontal": "center",
          "text_alignment_vertical": "center",
          "text_color": "#fff",
          "type": "text",
          "width": Object {
            "type": "fixed",
            "value": 135,
          },
        },
        Object {
          "font_size": 32,
          "font_weight": "medium",
          "line_height": 40,
          "margins": Object {
            "top": 24,
          },
          "text": "Что-то пошло не так",
          "text_alignment_horizontal": "center",
          "text_color": "#80ffffff",
          "type": "text",
        },
        Object {
          "font_size": 28,
          "font_weight": "medium",
          "line_height": 36,
          "margins": Object {
            "top": 8,
          },
          "text": "Проверьте интернет или обновите экран",
          "text_alignment_horizontal": "center",
          "text_color": "#80ffffff",
          "type": "text",
        },
      ],
      "type": "container",
    },
    Object {
      "delimiter_style": Object {
        "color": "#0fff",
      },
      "height": Object {
        "type": "match_parent",
        "weight": 3,
      },
      "type": "separator",
    },
  ],
  "paddings": Object {
    "bottom": 24,
    "left": 24,
    "right": 24,
    "top": 24,
  },
  "type": "container",
  "width": Object {
    "type": "match_parent",
    "weight": 1,
  },
}
`;
